### PR TITLE
Batool_design_browser_history

### LINKED ISSUE #47: Python Challenge: Brain teasers > Design browser history

### PR DESCRIPTION
**Pull request by BatoolKad**
- [ ] PR's title includes the challenge's name and language
- [ ] PR has short and clear description of the challenge
- [x] PR has appropriate labels and milestones for easy identification
- [x] PR it is assigned to the owner
- [x] reviewers are assigned
- [x] the PR contributes only one focused change
- [ ] the branch is up to date with `main`/`master`
- [x] the code works when pulled and run locally
- [ ] all conflicts are resolved (if any)
- [x] It is linked to an issue in the appropriate column of the project board
- [ ] feedback is addressed (if any, and if it is appropriate feedback.)

## README Documentation

- [x] The solution is documented in `/src/README.md`
- [x] The markdown source is formatted
- [x] Spelling and grammar is correct in all text
- [ ] The markdown looks correct when you preview the file
- [x] All links and images work
- [x] The README documents the solution's behavior, strategy(ies) and implementation(s)
- [x] There are use cases in code blocks to illustrate the function's behavior

## Python Files

- [ ] There is a module header
- [ ] There is a module docstring
- [x] File names are in snake_case
- [x] Test files are named `test_<module_name>.py`

## Function Docstring

- [x] Behavior description
- [x] Parameter description
- [x] Return value description
- [x] Include any assumptions
- [ ] Include 3 or more (passing!) doctests
- [x] Include 1-2 use cases (if necessary)

## Function Implementation

- [x] The solution is not is not copied from [djeada](https://github.com/djeada/Algorithms-And-Data-Structures)
- [ ] The code is auto-formatted
- [x] The code has no (reasonable) linting mistakes
- [x] Variables are named with snake_case
- [x] The function has a clear and helpful name
- [x] The file's name matches the function name
- [x] The code follows the strategy as simply as possible
- [x] Variable names are clear and helpful
- [x] Comments explain the strategy (if necessary)
- [x] There are type annotations
- [x] (challenge) The code includes defensive assertions

## Unit Test Suite

- [x] The test class has a helpful name in PascalCase
- [ ] The test class has a docstring
- Each unit test has
  - [ ] A helpful name
  - [ ] A clear docstring
  - [ ] Only one assertion
- [x] All tests pass
- [ ] There are tests for defensive assertions
- [ ] There are tests for boundary cases
- [ ] Test suite includes black-box and glass-box tests

### CODE REVIEW

File: src/brain_teasers/python/deletions_to_make_valid_parentheses/tests/test_deletions_to_make_valid_parentheses.py [hunk 1]
```diff
@@ -1,8 +1,12 @@
 import unittest
 
-from src.deletions_to_make_valid_parentheses import (
```
Comment (abed42) [inline src/brain_teasers/python/deletions_to_make_valid_parentheses/tests/test_deletions_to_make_valid_parentheses.py:3]: did you try to include  this directory in the PYTHONPATH environment ? there are some solutions here : [stackoverflow](https://stackoverflow.com/questions/51049663/python3-6-error-modulenotfounderror-no-module-named-src)

File: src/brain_teasers/python/design_browser_history/README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
+
+---
+##BrowserHistory Class
```
Comment (abed42) [inline src/brain_teasers/python/design_browser_history/README.md:3]: minor issue. This title isn't showing up formatted because you need space between the hashtags and the title. like  : `## BrowserHistory Class` instead of `##BrowserHistory Class`

Comment (abed42) [reply src/brain_teasers/python/design_browser_history/README.md:3]: other than that LGTM ( === looks good to me)

File: src/brain_teasers/python/design_browser_history/src/design_browser_history.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    def forward(self, step: int) -> str:
+        """
+        :param step:the forward steps
+        :return:string - current url webpage after moving forward
+        """
+        # move forward # steps, limited by the last index
+        self.index = min(len(self.urls_history)-1, self.index + step)
+
+        ## return the new webpage url
+        return self.urls_history[self.index]
```
Comment (abed42) [inline src/brain_teasers/python/design_browser_history/src/design_browser_history.py:51]: LGTM

**Review by abed42 — CHANGES_REQUESTED** [review]
@BatoolKad almost ready to merge, just please fix the typo in the readme.
